### PR TITLE
Use bootstrap modal instead of confirm

### LINKED ISSUE #2667: Use Bootstrap 4 modals instead of javascript confirm()
Everywhere in the admin UI when we need to delete a content item we should be using BS 4 modals.

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #2667

### CODE REVIEW

**Comment by Skrypt** [comment]
This I approve 😉

**Comment by sebastienros** [comment]
Shouldn't Gulp run to generate the final assets ?

**Comment by hishamco** [comment]
Ya .. I really forgot

**Comment by Skrypt** [comment]
Do it do it !! 😋

**Comment by hishamco** [comment]
Here we are .. I just delayed little bit because I was on my mobile, opening my laptop, run the gulp task and push the commit ..

File: src/OrchardCore.Themes/TheAdmin/wwwroot/Scripts/TheAdmin.js [hunk 1]
```diff
@@ -6529,29 +6529,74 @@ return Popper;
...
+                <div class="modal-body">
+                    <p>`+ message + `</p>
+                </div>
+                <div class="modal-footer">
+                    <button id="modalOkButton" type="button" class="btn btn-danger">`+ $('#confirmRemoveModalMetadata').data('ok') + `</button>
+                    <button id="modalCancelButton" type="button" class="btn btn-secondary" data-dismiss="modal">`+ $('#confirmRemoveModalMetadata').data('cancel') + `</button>
+                </div>
+            </div>
+        </div>
+    </div>`).appendTo("body");
```
Comment (sebastienros) [inline src/OrchardCore.Themes/TheAdmin/wwwroot/Scripts/TheAdmin.js]: What happens if I click cancel and open the dialog again? Does it add this element multiple times with the same id?

**Comment by hishamco** [comment]
Nope .. it removes the entire modal

**Comment by Skrypt** [comment]
image
After clicking on unpublish it seems that the message is not properly passed to the modal.

**Comment by hishamco** [comment]
Let me check .. BTW we need to set the proper message text in many places

**Comment by deanmarcussen** [comment]
@hishamco @Skrypt The unpublish message was never set I think. #3444 addresses this. I will pull dev in tonight and check that it still works

**Comment by hishamco** [comment]
@deanmarcussen I saw your PR yesterday, could you please fetch the latest bits and rebase .. and check if this work with the new bootstrap modal

**Comment by deanmarcussen** [comment]
@hishamco #3444 works, but doesn't bring the title across as well, as that was never an attribute on the previous alert, so I will update to include title and push again

**Pull request merged**